src/main/java/com/xjeffrose/chicago/ChicagoObjectDecoder.java

```diff
@@ -31,10 +36,32 @@ public ChicagoMessage decode(byte[] msg) {
   @Override
   protected void decode(ChannelHandlerContext ctx, ByteBuf msg, List<Object> out) throws Exception {
     // Populate the output List
+    List<ChicagoMessage> messages = new ArrayList<>();
 
     while (msg.readableBytes() > 50) {
       out.add(_decode(msg));
```

Comment:
was this meant to be a call to `messages`?  That would avoid the casting in the if below as well as the call to `out.clear`